Runtime/JSNativeApi.cs

```diff
@@ -362,12 +362,14 @@ public static bool InstanceOf(this JSValue thisValue, JSValue constructor)
         }
     }
 
-    public static unsafe JSValue DefineClass(ReadOnlySpan<byte> utf8Name, JSCallback callback, params JSPropertyDescriptor[] descriptors)
+    public static unsafe JSValue DefineClass(
+        ReadOnlySpan<byte> utf8Name,
+        JSCallbackDescriptor constructorDescriptor,
+        params JSPropertyDescriptor[] propertyDescriptors)
     {
-        int nameLength = utf8Name.Length;
-        GCHandle callbackHandle = GCHandle.Alloc(callback);
+        GCHandle callbackHandle = GCHandle.Alloc(constructorDescriptor);
```

Comment:
>GCHandle.Alloc(constructorDescriptor); [](http://example.com/codeflow?start=34&length=38)

How does it work for structs?

Reply:
I see that we do the same for the `JSPropertyDescriptor`. I guess it is being boxed.

Reply:
Yes, it gets boxed since `GCHandle.Alloc()` takes an `object`. Then it's unboxed by casting the callback args `object data` value to `JSCallbackDescriptor`.

`JSPropertyDescriptor` is not quite the same because it gets copied into an array of `napi_property_descriptor`.